### ISSUE TITLE
What about directx implemention? what about using SDL and bgfx ?

### ISSUE THREAD

**Issue by meiry** [axys1/axys#783]
Hey 
Any planes to Implement directx for windows ?
Or replace the render layer with something like bgfx ? and maybe the windowing with SDL2 
Now that its only on a voluntary basis you will have hard time to maintain diffrente parts of the engine

**Comment by aismann** [comment]
see here:
Render engine collection #67

**Comment by meiry** [comment]
Thanks , ok so i saw some half baked and some expiriments , 
But can you please tell me who is the leader of this project ? who decides what will get in in this fork?

**Comment by DelinWorks** [comment]
It's @halx99 he is the leader of this repo

**Comment by DelinWorks** [comment]
> Hey Any planes to Implement directx for windows ?
I don't know if you know already but there is a good translation layer called `Google Angle` that's good to some extent! you can enable it with `AX_USE_COMPAT_GL`

**Comment by meiry** [comment]
its not optimized , its like you said translation layer , im talking to more direct approach

**Comment by DelinWorks** [comment]
> its not optimized , its like you said translation layer , im talking to more direct approach

yeah I get what you mean.. it sucks tbh because it uses an opengl es mobile backend which is counter-intuitive for performance, also shaders don't work right away and sometimes it crashes on my GPU for no reason

**Comment by DelinWorks** [comment]
https://github.com/google/filament
https://github.com/OGRECave/ogre-next
https://github.com/DiligentGraphics/DiligentEngine

These top rendering repos seem to focus on 3D and Physically based rendering which this engine isn't intended for unless you guys want it to be a 2D/3D engine combo ;)

but this https://github.com/bkaradzic/bgfx seems like it wraps OpenGL, DirectX and metal which is what we are looking for, with a bonus of 3D shaders like deferred lighting and basic shadows, toon shading and whatnot

**Comment by aismann** [comment]
> but this https://github.com/bkaradzic/bgfx seems like it wraps OpenGL, DirectX, Vulkan and Metal which is what we are looking for, with a bonus of 3D shaders like deferred lighting and basic shadows, toon shading and whatnot, also I used this library before and the API seems to be robust

I think also BGFX is the right one.

**Comment by meiry** [comment]
The problem is that this implemention to be production ready on all combos(pc/mobile/web/mac/linux)  is a lot of work . 
Its a shame to take production level engine and abandon it . it will unfortunately toy project . 
i know this engine for 10 years .  
Any good luck to all

**Comment by aismann** [comment]
@meiry 
What are you want?
In your first post you ask about SDL/ native DirectX support.
We be happy to get your PRs

**Comment by aismann** [comment]
@halx99 
Please close it.
Seems to be there was a troll here.

**Comment by meiry** [comment]
no sorry , not troll at all , i just wanted to know the state of this project ,i go tthe general idea . 
Its important to others to know also.

**Comment by aismann** [comment]
@meiry 
Axys is on Beta stage.

**Comment by aismann** [comment]
@halx99 
Please close it.

**Comment by DelinWorks** [comment]
@meiry the imlementation is great because it wraos all the rendering apis in simple commands, and it shouldn't be hard to implement, also this project is not a toy project and it's a professional production ready project just like the tool to be implemented.

**Comment by DelinWorks** [comment]
also why do you want d3d11 support? not like this engine is a 3D realistic game engine like unreal engine, if anything we could just switch to vulkan just like godot did, never gave a shit about that closed api called directx tbh

**Comment by DelinWorks** [comment]
@meiry i see what you mean and i don't consider you a troll in my book :)

**Comment by DelinWorks** [comment]
So i've been thinking about it and we can have single support for Vulkan and that's it, as for IOS devices we could use [MoltenVK](https://moltengl.com/moltenvk/)
OR we could use BGFX!
Thanks to apple now we'd need to rewrite the rendering api twice if we're going to implement vulkan and translate it to their stupid metal API

**Comment by meiry** [comment]
@DelinWorks  , 
For you questions : Directx is very important for desktop and xbox .  OpenGL in the end is not native on windows OS . 
The Directx question was rasied over and over again in the cocos2d-x days . but never got the time to implement . 
Now for teachnical side . i think that time wise bgfx is the shortest way to get them all . 
But i dont know if its optimal for games .

**Comment by DelinWorks** [comment]
it is optimal! commercial companies built their engines on top of it.
also to make games for Xbox you'd need UWP platform support in the first place and then you'd need a DirectX implementation.

**Comment by halx99** [comment]
it's ok

**Comment by meiry** [comment]
@DelinWorks , 
What commercial companies  build there games around bxfg except wha in the bxfg site ?

**Comment by DelinWorks** [comment]
What do you underestimate about this tool? it works great it supplies a great interface to all APIs... not like it's gonna be more profissional than this project.

**Comment by meiry** [comment]
i do not underestimate  at all just asking questions

**Comment by aismann** [comment]
Should be close.
All answers can be found on readme and included links.

**Comment by DelinWorks** [comment]
> i do not underestimate at all just asking questions

or we can implement Vulkan only and then translate it using ANGLE for Xbox. I think that's a solution.

**Comment by aismann** [comment]
@DelinWorks, 
We have a roadmap. 
@meiry Look at roadmap plz.

**Comment by DelinWorks** [comment]
Do you mean milestones?

**Comment by aismann** [comment]
> Do you mean milestone.
Yes.

**Comment by DelinWorks** [comment]
so what in there?

**Comment by DelinWorks** [comment]
ohh you mean the `1.1` milestone. I get it we should close this and mark it in there

**Comment by rh101** [comment]
This should be closed, because it's not an issue or a problem with the engine, it's more of a discussion, should be moved to the [Discussions](https://github.com/axys1/axys/discussions) section of this project.
